src/main/java/com/minecolonies/coremod/entity/ai/citizen/farmer/EntityAIWorkFarmer.java

```diff
@@ -391,14 +397,66 @@ private IAIState canGoPlanting(@NotNull final ScarecrowTileEntity currentField,
      *
      * @param position the position to check.
      * @param field    the field close to this position.
-     * @return true if should be hoed.
+     * @return position of hoeable surface or null if there is not one
      */
-    private boolean shouldHoe(@NotNull final BlockPos position, @NotNull final ScarecrowTileEntity field)
+    private BlockPos findHoeableSurface(@NotNull BlockPos position, @NotNull final ScarecrowTileEntity field)
     {
+        position = getSurfacePos(position);
+        if (position == null
+            || field.isNoPartOfField(world, position) 
+            || (world.getBlockState(position.above()).getBlock() instanceof CropsBlock)
+            || (world.getBlockState(position.above()).getBlock() instanceof BlockScarecrow)
+            || !world.getBlockState(position).getBlock().is(Tags.Blocks.DIRT)
+        )
+        {
+            return null;
+        }
+        return position;
+    }
 
-        return !field.isNoPartOfField(world, position) && !(world.getBlockState(position.above()).getBlock() instanceof CropsBlock)
-                 && !(world.getBlockState(position.above()).getBlock() instanceof BlockScarecrow)
-                 && (world.getBlockState(position).getBlock().is(Tags.Blocks.DIRT) || world.getBlockState(position).getBlock() instanceof GrassBlock);
+    /**
+     * Finds the position of the surface near the specified position
+     *
+     * @param position the location to begin the search
+     * @return the position of the surface block or null if it can't be found
+     */
+    private BlockPos getSurfacePos(final BlockPos position)
+    {
+        return getSurfacePos(position, 0);
+    }
+
+    /**
+    * Finds the position of the surface near the specified position
+    *
+    * @param position the location to begin the search
+    * @param depth the depth of the search for the surface
+    * @return the position of the surface block or null if it can't be found
+    */
+    private BlockPos getSurfacePos(final BlockPos position, final Integer depth)
+    {
+        if (Math.abs(depth) > MAX_DEPTH)
+        {
+            return null;
+        }
+        final BlockState curBlockState = world.getBlockState(position);
+        @Nullable final Block curBlock = curBlockState.getBlock();
+        if (curBlock.is(Tags.Blocks.DIRT) || curBlock.is(Tags.Blocks.STONE) || curBlockState.getMaterial().isLiquid() || ((curBlock instanceof IWaterLoggable) && curBlockState.getValue(BlockStateProperties.WATERLOGGED)) || curBlock instanceof FarmlandBlock)
```

Comment:
Still, stone tag is not any useful, check for solid material rather then.

Reply:
I was trying to emulate the current behavior as closely as possible. Currently, if you have wood or crafted stone on top of the field, the farmer will break it in order to till the field. I'm happy to update it to not do that anymore, though. That would allow farms to butt up against buildings without damaging them, too.

Reply:
I'm out right now, but when I get home I'll update it to look for solid blocks, liquids, or waterlogged blocks